File: app/src/main/java/com/example/blogmusic/ui/components/PostDetail.java
package com.example.blogmusic.ui.components;

import java.util.List;
import java.util.Map;

public class PostDetail {
    private final int post_id;
    private final String title;
    private final String author;
    private final String date;

    private final String subtitle;
    private final String introduction;
    private final String main_content;
    private final String conclusion;
    private final String tags;
    private final String image_cover;
    private final Map<String, List<Media>> media;

    public PostDetail(int post_id, String title, String author, String date, String subtitle, String introduction, String mainContent, String conclusion, String tags, String image_cover, Map<String, List<Media>> media) {
        this.post_id = post_id;
        this.title = title;
        this.author = author;
        this.date = date;
        this.subtitle = subtitle;
        this.introduction = introduction;
        this.main_content = mainContent;
        this.conclusion = conclusion;
        this.tags = tags;
        this.image_cover = image_cover;
        this.media = media;
    }

    public int getPostId() { return post_id; }
    public String getTitle() { return title; }
    public String getAuthor() { return author; }
    public String getDate() { return date; }
    public String getSubtitle() { return subtitle; }
    public String getIntroduction() { return introduction; }
    public String getMainContent() { return main_content; }
    public String getConclusion() { return conclusion; }
    public String getTags() { return tags; }
    public String getImageCover() {return image_cover;}
    public Map<String, List<Media>> getMedia() { return media; }
}

File: blog_music_api/post/post_detail.php
<?php
include '../db.php';
header('Content-Type: application/json');

if ($_SERVER['REQUEST_METHOD'] === 'GET' && isset($_GET['id'])) {
    $id = intval($_GET['id']);

    // Lấy thông tin chi tiết của bài viết
     $stmt = $conn->prepare("
        SELECT p.title, p.author, p.date, p.image_cover, 
        d.post_id, d.subtitle, d.introduction, d.main_content, d.conclusion, d.tags
        FROM posts p
        JOIN post_detail d ON p.id = d.post_id
        WHERE p.id = ?" );
    $stmt->bind_param("i", $id);
    $stmt->execute();
    $result = $stmt->get_result();

    if ($post = $result->fetch_assoc()) {
        // Lấy media liên quan đến bài viết và nhóm theo position
        $mediaStmt = $conn->prepare("SELECT file_url, type, position FROM media WHERE post_id = ?");
        $mediaStmt->bind_param("i", $id);
        $mediaStmt->execute();
        $mediaResult = $mediaStmt->get_result();

        $mediaByPosition = [
            'subtitle' => [],
            'introduction' => [],
            'main_content' => [],
            'conclusion' => [],
            'tags' => []
        ];

        while ($m = $mediaResult->fetch_assoc()) {
            $position = $m['position'];
            if (isset($mediaByPosition[$position])) {
                $mediaByPosition[$position][] = [
                    'file_url' => $m['file_url'],
                    'type' => $m['type']
                ];
            }
        }
        $post['media'] = $mediaByPosition;
        echo json_encode($post, JSON_PRETTY_PRINT | JSON_UNESCAPED_UNICODE);
    } else {
        echo json_encode(["error" => "Post not found"]);
    }
} else {
    echo json_encode(["error" => "Invalid request"]);
}
?>
